src/controllers/localWebService.ts

```diff
@@ -43,8 +48,10 @@ export default class LocalWebService {
         this.server.on('request', this.app);
 
         // Handle new connections to the web socket server
-        this.wss.on('connection', (ws) => {
-            let parse: any = querystring.parse(<string>url.parse(ws.upgradeReq.url).query);
+        this.wss.on('connection', (ws, req) => {
+            // upgradeReq was removed from the API
+            // See - https://github.com/websockets/ws/pull/1099
+            let parse: any = querystring.parse(<string>url.parse(req.url).query);
 
```

Comment:
Was this actually broken? Or did you just happen to see that it was removed and did this now so that if we updated the version later on it wouldn't break? Because right now it seems like this isn't actually changing/fixing anything

Reply:
No it was broken in master when I started working on this. And I saw the last commit was bumping the version for this library.

Reply:
Alright. I'd just remove the comment though. It has no context here unless you know the history (that it used to be ws.upgradeReq). The logic just using req.url seems fine by itself - not sure why it was using upgradeReq in the first place.

---
In reply to: [303567328](https://github.com/microsoft/vscode-mssql/pull/1259#discussion_r303567328) [](ancestors = 303567328)